aiographql/client/client.py

```diff
@@ -92,118 +91,182 @@ def __init__(
 
     async def validate(
         self,
-        query: str,
+        request: GraphQLRequest,
         schema: Optional[graphql.GraphQLSchema] = None,
         headers: Optional[Dict[str, str]] = None,
-    ) -> List[graphql.GraphQLError]:
-        return await asyncio.get_running_loop().run_in_executor(
-            None,
-            graphql.validate,
-            schema or await self.get_schema(headers=headers),
-            graphql.parse(query),
+        force: bool = False,
+    ) -> None:
+        """
+        Validate a given request against a schema (provided or fetched). Validation is
+        skipped if the request's `validate` property is set to `False` unless forced.
+
+        :param request: Request that is to be validated.
+        :param schema: Schema against which provided request should be validated, if
+                       different from `GraphQLRequest.schema` or as fetched from the
+                       client endpoint.
+        :param headers: Headers to be set when fetching the schema from the client
+                        endpoint. If provided, request headers are ignored.
+        :param force: Force validation even if the provided request has validation
+                      disabled.
+        """
+        if not request.validate and not force:
+            # skip validation if request validate flag is set to false
+            return
+
+        if request.schema is None:
+            request.schema = schema or await self.get_schema(
+                headers=headers or request.headers
+            )
+
+        errors = await asyncio.get_running_loop().run_in_executor(
+            None, graphql.validate, request.schema, graphql.parse(request.query)
         )
+        if errors:
+            raise GraphQLClientValidationException(*errors)
 
-    async def _validate(
-        self, request: GraphQLRequest, headers: Optional[Dict[str, str]] = None
-    ):
-        if request.validate:
-            errors = await self.validate(request.query, request.schema, headers=headers)
-            if request.schema is None:
-                request.schema = await self.get_schema(headers=headers)
-            if errors:
-                raise GraphQLClientValidationException(*errors)
+    def _prepare_request(
```

Comment:
The main reason why I went with this approach is to enable the user to not require a `GraphQLRequest` object when making a request. So for the simplest case,

```py
response = client.request("<query string>")
```

Instead of,
```py
response = client.request(GraphQLRequest("<query string>"))
```

But to answer your inital question, yes the user can call the request method explicitly as well.